### PR TITLE
Set up framework for handling questionnaire data

### PR DESCRIPTION
**Pull request by rdgoite**
* Restructured the code for downloading the dummy spreadsheet to use service that's intended to invoke the template generator backend.
* Initially defined framework for processing form data into template specification.

### CODE REVIEW

**Comment by rolando-ebi** [comment]
@rdgoite looks good so far, but also requires the mechanism whereby the broker's HTTP API responds with a URL for the asynchronously generated spreadsheet. The UI should then poll that URL until the spreadsheet file is available to download

**Comment by rdgoite** [comment]
While the framework is built for now with the assumption that the spreadsheet will be (almost) instantly ready, the services and components can be modified to to accommodate async generation.

However, from a usability perspective, how would that work? Do we need something like a modal box to tell the user that we're generating the template while we poll the URL in the background?

**Comment by rolando-ebi** [comment]
Yeah, I was under the impression we were going to have UI mock-ups for a spinny loading wheel or something like that while the spreadsheet is generated in the background

**Comment by rolando-ebi** [comment]
I asked Clare about this. She said there are no mockups, but for now we should just change the "generate" button to "generating" while we're awaiting the spreadsheet blob

**Comment by MightyAx** [comment]
@rdgoite @rolando-ebi: we do have spinny wheel that we can show while something takes some time, check out the loader service here: https://github.com/ebi-ait/ingest-ui/blob/3c619f8e4446cdfe5f8949527f667a7747cc3db9/client/src/app/project/project.component.ts#L122

File: client/src/app/template-questionnaire/template-questionnaire.data.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export interface QuestionnaireData {
```
Comment (MightyAx) [inline client/src/app/template-questionnaire/template-questionnaire.data.ts:1]: This doesn't account for the full schema of info specified [here](https://github.com/ebi-ait/ingest-ui/blob/master/client/src/app/template-questionnaire/template-questionnaire-form/template-questionnaire-schema.json)

Also how does this account for the schema changing? (as @ami-day is currently doing)

**Pull request merged**